### ISSUE TITLE
Boolean variants don't match false

### ISSUE THREAD

**Issue by rafgraph** [modulz/stitches#292]
# Bug report

## Describe the bug

Boolean variants don't match `false`

```js
const Result = styled("div", {
  color: "blue",
  variants: {
    bool: {
      true: {
        color: "green"
      },
      false: {
        color: "red"
      }
    }
  }
});

// This works and is green
<Result bool={someTrueVariable}>True</Result>

// This does not work and ends up blue
<Result bool={someFalseVariable}>False</Result>

// This throws a typescript error
<Result bool={someFalseVariable.toString()}>False</Result>

// This is the only way that works
<Result bool={someBooleanVariable ? 'true' : 'false'}>...</Result>
```

## To Reproduce

https://codesandbox.io/s/abc-platform-vg40r-q9lsh?file=/src/App.tsx:82-259

## Expected behavior

For the false variant to match, and for `someBooleanVariable.toString()` to not throw a typescript error.

**Comment by rafgraph** [comment]
To give some real world context to this issue, I came across it when creating the demo site for [Detect It](https://github.com/rafgraph/detect-it) and styling the detection results.

As a side note, this was first time using Stitches and I really enjoyed it, excited to see where it goes!

**Comment by peduarte** [comment]
@rafgraph hey, thanks! This has been fixed and released under the `canary` flag. 

I recommend upgrading to `0.0.3-canary.4` as it contains a bunch of fixes since `0.0.2`. For more information, refer to the [releases page](https://github.com/modulz/stitches/releases).

Please note that there are some small breaking changes, as noted in the release logs.

Thanks, any questions feel free to ask on here or on our [Discord channel](https://discord.com/invite/H4eG3Mk).

**Comment by rafgraph** [comment]
Thanks @peduarte, I tried upgrading to `0.0.3-canary.4` and I updated `css.global` for the breaking change, but my dark mode theme breaks. It's weird because most of the tokens change values, but not `$text` and `$textLowContrast`. The attempted upgrade is here (it's a fairly basic site): https://codesandbox.io/s/github/rafgraph/detect-it-demo/tree/stitches-0.0.3-canary.4?file=/src/stitches.config.ts, for comparison it should look like https://detect-it.rafgraph.dev. The only changes made between the two were for the `css.global` breaking change, [diff here](https://github.com/rafgraph/detect-it-demo/commit/6cb044eb5efcd15b012a0b7e63a97054322ea5fd). Maybe I missed another breaking change in the change log?
